File: Chess Backend/Models/MoveResult.cs
namespace Chess_Backend.Models
{
    public class MoveResult
    {
        public bool IsValidMove { get; set; }
        public string? Message { get; set; }
        public bool IsCheck { get; set; }
        public bool IsCheckmate { get; set; }
        public bool IsStalemate { get; set; }
        public Piece? CapturedPiece { get; set; }
        public bool IsDraw { get; set; }
        public bool IsPromotion { get; set; }
        public bool IsEnPassant { get; set; }
        public bool IsCastling { get; set; }
    }
}

File: chessUI/src/models/MoveResult.ts
import { Piece } from "../types";

export interface MoveResult {
    isValidMove: boolean;
    message: string;
    isCheck: boolean;
    isCheckmate: boolean;
    isStalemate: boolean;
    capturedPiece: Piece[] | null;
    isDraw: boolean;
    isPromotion: boolean;
    isEnPassant: boolean;
    isCastling: boolean;
}